Update dstack-proxy contributing guide

jvstme opened the pull request:
Closes #2252

jvstme committed:
Update dstack-proxy contributing guide

jvstme commented on contributing/PROXY.md:
[Rendered](https://github.com/dstackai/dstack/blob/issue_2252_proxy_contributing_guide/contributing/PROXY.md)

r4victor commented on contributing/PROXY.md:
````diff
@@ -0,0 +1,194 @@
+   ssh -i /path/to/the/gateway/key ubuntu@gateway.example
+   ```
+
+1. Prepare an environment with your development branch on the gateway:
````
I wish the recommended approach wouldn't involve pushing changes to a remote branch. Previously, the guide suggested ssh/scp. Not ideal, but do you think this one is better?

jvstme replied:
The previous guide suggested building and uploading a wheel. I think running from source is preferable, as it allows to iterate faster, run a debugger, and possibly edit the code on the gateway.

I've tried using rsync to upload the source code, but it's difficult to set up to upload a correct copy of the repo without uploading unnecessary files, like the local venv, and without deleting the necessary remote files, like the remote venv.

But I can include an example on how to push the code via git without the GitHub repo.

jvstme committed:
Example on pushing without GitHub

Also drop the dependency on `build`, since the
guide no longer suggests building wheels
